### PR TITLE
Handling # after exponential notation

### PR DESCRIPTION
**Pull request by nguhoa**
Fix issue: https://github.com/microsoft/Power-Fx/issues/1946

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by LucGenetier — APPROVED** [review]
:shipit:

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text_ExcelCompat.txt [hunk 1]
```diff
@@ -2824,4 +2824,456 @@ Error({Kind:ErrorKind.InvalidArgument})
...
+"12345"
+
+>> Text(12345.34567,"##;(0.00E+00##)")
+"12345"
+
+>> Text(-12345.34567,"##;(0.00E+00##)")
+"(1.23E+004)"
+
+>> Text(0,"(0.00E+0#);##;(0.00E+000###)")
+"(0.00E+0000)"
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text_ExcelCompat.txt:2991]: >0000 [](http://example.com/codeflow?start=8&length=4)

Why is it 4 zeros? The format only specifies 3 required places, and the value (0) is not larger than that.

Comment (CarlosFigueiraMSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text_ExcelCompat.txt:2991]: Same issue in many other test cases

Comment (nguhoa) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text_ExcelCompat.txt:2991]: This is how excel does. # after 0 count as another 0. Can you recheck ?

Comment (CarlosFigueiraMSFT) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text_ExcelCompat.txt:2991]: Oh yeah, Excel is weird :-|... I guess that's right then :-)

**Pull request merged**